src/components/Popups/ReplyField/ReplyField.tsx

```diff
@@ -50,6 +69,47 @@ export default class ReplyField extends React.Component<Props, State> {
         this.saveCursorPosition();
     }
 
+    getVirtualElement = (activeMention: Mention): VirtualElement | null => {
+        const selection = window.getSelection();
+        if (!selection?.focusNode) {
+            return null;
+        }
+
+        const range = selection.getRangeAt(0);
+        const textNode = range.endContainer;
+
+        range.setStart(textNode, activeMention.start);
+        range.setEnd(textNode, activeMention.start);
+
+        const mentionRect = range.getBoundingClientRect();
+
+        // restore selection
+        range.setStart(textNode, activeMention.end);
+        range.setEnd(textNode, activeMention.end);
+
+        return {
+            getBoundingClientRect: () => mentionRect,
+        };
+    };
+
+    handleChange = (nextEditorState: EditorState): void => {
+        const { onChange, onMention } = this.props;
+
+        onChange(nextEditorState.getCurrentContent().getPlainText());
+
+        const activeMention = getActiveMentionForEditorState(nextEditorState);
+
+        if (onMention && activeMention) {
+            onMention(activeMention.mentionString);
+        }
+
+        this.setState({ editorState: nextEditorState }, () => {
+            // getVirtualElement has to wait for editorState update
+            const popupReference = activeMention ? this.getVirtualElement(activeMention) : null;
+            this.setState({ popupReference });
```

Comment:
This will trigger two renders. Less state is almost always better, unless it introduces a performance issue. Can we derive these values in render, instead?

```
render() {
    const { className, isDisabled, placeholder, ...rest } = this.props;
    const { editorState } = this.state;
    const activeMention = getActiveMentionForEditorState(editorState);
    const popupReference = activeMention ? this.getVirtualElement(activeMention) : null;

    return (
        <div className={classnames(className, 'ba-ReplyField')}> // ...etc
```